lib/parallel_test_base.pm

```diff
@@ -0,0 +1,1946 @@
+            $_ret = script_run("ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root\@$_peer groupmod -g $_group{$_single_group} $_single_group");
+            save_screenshot;
+            croak("$_single_group GID modification failed on $_peer") if ($_ret != 0);
+        }
+    }
+}
+
+=head2 config_host_shared_storage
+
+Configure shared nfs storage on server and mount it on client. Main arguments
+are type of shared storage, path of exported shared storage, server or client 
+and mount path on client. 
+=cut
+
+sub config_host_shared_storage {
+    my ($self, %args) = @_;
+    $args{type} //= 'nfs';
+    $args{exppath} //= '/var/lib/libvirt/images';
+    $args{role} //= 'server';
+    $args{mntpath} //= '/var/lib/libvirt/images';
+
+    record_info("Configure host shared storage");
+    if ($args{type} eq 'nfs') {
+        my $_temppath = get_var('EXTERNAL_NFS_SHARE', '');
+        if ($_temppath) {
+            script_run("umount $args{mntpath} || umount -f -l $args{mntpath}");
+            assert_script_run("mount -t nfs $_temppath $args{mntpath}");
+            $args{role} eq 'server' ? assert_script_run("rm -f -r $args{mntpath}/nfsok; touch $args{mntpath}/nfsok") : assert_script_run("cd ~ && ls -lah $args{mntpath}/nfsok");
+        }
+        elsif ($args{role} eq 'server') {
+            $_temppath = $args{exppath};
+            $_temppath =~ s/\//\\\//g;
+            assert_script_run("sed -i \'/^.*$_temppath.*\$/d\' /etc/exports");
+            assert_script_run("echo \"$args{exppath} *(rw,sync,no_root_squash,no_subtree_check)\" >> /etc/exports");
+            assert_script_run("exportfs -a");
+            systemctl('restart nfs-server.service');
+            systemctl('status nfs-server.service');
+            assert_script_run("touch $args{exppath}/nfsok");
+            save_screenshot;
+        }
+        elsif ($args{role} eq 'client') {
+            my $_nfsserver = get_var('PEER_IPADDR');
+            script_run("umount $args{mntpath} || umount -f -l $args{mntpath}");
+            assert_script_run("mount -t nfs $_nfsserver:$args{exppath} $args{mntpath}");
+            assert_script_run("cd ~ && ls -lah $args{mntpath}/nfsok");
+            save_screenshot;
+        }
+    }
+}
+
+=head2 config_host_security
+
+Get rid of limitations come from security services and rules which may have impact
+on connectivity between host and guest.
+=cut
+
+sub config_host_security {
+    my $self = shift;
+
+    record_info("Config host security");
+    my @_security_service = ('SuSEFirewall2', 'firewalld', 'apparmor');
+    foreach my $_ss (@_security_service) {
+        if (script_run("systemctl is-enabled $_ss") == 0) {
+            systemctl("stop $_ss");
+            systemctl("disable $_ss");
+            save_screenshot;
+        }
+    }
+
+    if (script_run("ls /etc/selinux/config") == 0) {
+        assert_script_run("sed -i -r \'s/^SELINUX=.*\$/SELINUX=disabled/g\' /etc/selinux/config");
```

Comment:
This will require reboot to take effect...

Reply:
Do you want to remove this line ?

Reply:
It seems the existing host setup in developing and MU test is ok on this, and doesn't affect VT test.  For alp, selinux will be setup during host installation too. Migration test will be integrated after those steps, so should be fine on this , I think.

Reply:
I can still remove this I think anyway.